src/main/java/net/sf/jabref/logic/util/io/FileHistory.java

```diff
@@ -0,0 +1,89 @@
+/*  Copyright (C) 2003-2015 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package net.sf.jabref.logic.util.io;
+
+import java.util.LinkedList;
+
+import net.sf.jabref.JabRefPreferences;
+
+public class FileHistory {
+
+    private final JabRefPreferences prefs;
+    private final LinkedList<String> history = new LinkedList<>();
+
+
+    public FileHistory(JabRefPreferences prefs) {
+        this.prefs = prefs;
+        String[] old = prefs.getStringArray(JabRefPreferences.RECENT_FILES);
+        if ((old != null) && (old.length > 0)) {
+            for (int i = 0; i < old.length; i++) {
```

Comment:
You can use for(String s : old) { list.add(s); } instead. The check for old.length > 0 is also not necessary as the loop would ignore the case in which the array is empty.